tasks/main.yml

```diff
@@ -51,44 +74,33 @@
     - "{{ clickhouse_path_tmp }}"
     - "{{ clickhouse_path_data }}"
 
-- name: Add clickhouse config
+- name: Add clickhouse config.d/ configs
   ansible.builtin.template:
-    src: config.j2
-    dest: "{{ clickhouse_path_configdir }}/config.xml"
+    src: "config.d/{{ item }}.j2"
+    dest: "{{ clickhouse_path_configdir }}/config.d/{{ item }}"
     owner: clickhouse
     group: clickhouse
+  with_items:
+    - logger.xml
+    - network.xml
+    - remote_hosts.xml
+    - dirs.xml
   notify:
     - restart clickhouse
 
-- name: Add clickhouse users config
+- name: Add clickhouse users.d/ configs
   ansible.builtin.template:
-    src: users.j2
-    dest: "{{ clickhouse_path_configdir }}/users.xml"
+    src: "users.d/{{ item }}.j2"
+    dest: "{{ clickhouse_path_configdir }}/users.d/{{ item }}"
```

Comment:
Noticed that we don't have a default for the `clickhouse_path_configdir` in here. As this role relies on it in multiple places it would be good to add a default value to the `defaults/main.yml`.